backend/api/parameter.proto

```diff
@@ -1,23 +0,0 @@
-// Copyright 2018 The Kubeflow Authors
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//      http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-syntax = "proto3";
-
-option go_package = "github.com/kubeflow/pipelines/backend/api/go_client";
-package api;
-
-message Parameter {
```

Comment:
This is also being used in the `pipeline.proto`, are we getting rid of the use of Parameters for good in v2 - what are we replacing it with?

Reply:
We are using the `Parameter` field inside `RuntimeConfig` only, not the one outside.

Reply:
I see what you are saying... Since the `Parameters` field in `RuntimeConfig` is in form of `map<string, google.protobuf.Value>` (see [link](https://github.com/kubeflow/pipelines/blob/35fccb1faecf3b6ab1edd4171885cfaaf4ed82ee/backend/api/pipeline_spec.proto#L48)), maybe we use the same format for all parameter fields?

Reply:
yeah I think that would work, sgtm

- [google.protobuf.Value](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Value)